src/main/io/osd.h

```diff
@@ -97,8 +97,32 @@ typedef enum {
     OSD_UNIT_METRIC
 } osd_unit_e;
 
+#define OSD_ORIGIN_MASK_N (1<<0)
+#define OSD_ORIGIN_MASK_E (1<<1)
+#define OSD_ORIGIN_MASK_S (1<<2)
+#define OSD_ORIGIN_MASK_W (1<<3)
+
+typedef enum {
+    OSD_ORIGIN_C  = 0,
+    OSD_ORIGIN_NW = OSD_ORIGIN_MASK_N | OSD_ORIGIN_MASK_W,
+    OSD_ORIGIN_N  = OSD_ORIGIN_MASK_N,
+    OSD_ORIGIN_NE = OSD_ORIGIN_MASK_N | OSD_ORIGIN_MASK_E,
+    OSD_ORIGIN_E  = OSD_ORIGIN_MASK_E,
+    OSD_ORIGIN_SE = OSD_ORIGIN_MASK_S | OSD_ORIGIN_MASK_E,
+    OSD_ORIGIN_S  = OSD_ORIGIN_MASK_S,
+    OSD_ORIGIN_SW = OSD_ORIGIN_MASK_S | OSD_ORIGIN_MASK_W,
+    OSD_ORIGIN_W  = OSD_ORIGIN_MASK_W
+} osd_origin_e;
+
+typedef struct {
+    int8_t x;
+    int8_t y;
+    osd_origin_e origin;
+} osdItemPos_t;
+
 typedef struct osdConfig_s {
-    uint8_t item_pos[OSD_ITEM_COUNT][2];
+    osdItemPos_t item_pos[OSD_ITEM_COUNT];
```

Comment:
Then this will need some renaming like this:
```
typedef struct {
    int8_t pos_x;
    int8_t pos_y;
    osd_origin_e origin;
    bool visible;
} osdItem_t;

typedef struct osdConfig_s {
    osdItem_t item[OSD_ITEM_COUNT];

    bool enabled_stats[OSD_STAT_COUNT];

    // Alarms
    uint8_t rssi_alarm;
    uint16_t cap_alarm;
    uint16_t time_alarm;
    uint16_t alt_alarm;

    osd_unit_e units;
} osdConfig_t;
```